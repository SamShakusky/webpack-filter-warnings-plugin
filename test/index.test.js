import FilterWarningsPlugin from '../src/cjs';
import webpack from './utils/webpack';

test('filter warnings by regexp', async () => {
  const stats = await webpack({
    extend: {
      plugins: [
        new FilterWarningsPlugin({ exclude: /hide/ }),
      ],
    },
  });
  expect(stats.compilation.warnings).toMatchSnapshot();
});

test('allow array of regexp to filter by', async () => {
  const stats = await webpack({
    extend: {
      plugins: [
        new FilterWarningsPlugin({ exclude: [/hide/] }),
      ],
    },
  });
  expect(stats.compilation.warnings).toMatchSnapshot();
});

test('throw when invalid arguments', async () => {
  try {
    await webpack({
      extend: {
        plugins: [
          new FilterWarningsPlugin({}),
        ],
      },
    });
  } catch (e) {
    expect(e).toMatchSnapshot();
  }
});
