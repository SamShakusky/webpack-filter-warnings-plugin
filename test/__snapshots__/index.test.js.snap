// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allow array of regexp to filter by 1`] = `
Array [
  [ModuleWarning: show me],
]
`;

exports[`filter warnings by regexp 1`] = `
Array [
  [ModuleWarning: show me],
]
`;

exports[`throw when invalid arguments 1`] = `[Error: Exclude an only be a regexp or an array of regexp]`;
